Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -297,14 +330,27 @@ private extension UInt8 {
         case .keepAlive:
             return true
         case .unknown:
+            var keepAlive: Bool = false
+            var close: Bool = false
             for header in self.headers {
+                print(self.buffer.getString(at: header.name.start, length: header.name.length))
+                print(self.buffer.getString(at: header.value.start, length: header.value.length))
                 if self.buffer.equalCaseInsensitiveASCII(view: "connection".utf8, at: header.name) {
-                    if self.buffer.equalCaseInsensitiveASCII(view: "close".utf8, at: header.value) {
-                        return false
-                    }
-                    return self.buffer.equalCaseInsensitiveASCII(view: "keep-alive".utf8, at: header.value)
+                    
+                    let (isThisKeepAlive, isThisClose) = HTTPKeepAliveHeader.parse(
+                        self.buffer, from: header.value.start, length: header.value.length)
+                    keepAlive = keepAlive || isThisKeepAlive
+                    close = close || isThisClose
                 }
             }
+            
+            if keepAlive && close {
+                fatalError("Couldn't be close and keep-alive at the same time")
```

Comment:
Errors like this at the protocol level must not be fatal: taking out the entire server due to one misbehaving client is not a good approach. In this circumstance we should probably default to choosing `close` over `keep-alive`.

Reply:
Sorry, didn't think about taking the server out.

Reply:
No need to apologise! Code review is not criticism, you don't need to feel like you made a mistake. We're working together to solve the problem. :smile: